File: 0035-search-insert-position/0035-search-insert-position.c
int searchInsert(int* nums, int numsSize, int target) {
    int i=0,index=0;;
    while(i<numsSize){
        if(nums[i]==target){
            return i;
        }

        if(nums[i]>target){
            return i;
        }
        i++;
    }
    return i;

}

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int left = 0, right = nums.length - 1;

        while(left <= right){
            int mid = left + (right - left) / 2;
            System.out.println(nums[mid]);
            if(nums[mid] == target) return mid;

            else if(nums[mid] > target) right = mid - 1;
            else left = mid + 1;
        }
        return left;
    }
}